### ISSUE TITLE
Can't use Types name without fully qualifying

### ISSUE THREAD

**Issue by TimDeve** [carp-lang/Carp#1183]
Tried to build an old project and it now fails because referring to types without fully qualifying them is not possible.

Use to work:
```clojure
(defmodule Mod
 (deftype Ok [a Int])
 (deftype NotOk [a Ok]))
```

What is needed now:
```clojure
(defmodule Mod
 (deftype Ok [a Int])
 (deftype NotOk [a Mod.Ok]))
```

**Comment by scolsen** [comment]
Thanks! This is due to the fact that we actually support module encapsulation of types now. Before the original code was really adding `Ok` and `NotOk` to the global type environment even though the calls occur in `Mod`, whereas now it actually adds it to `Mod`. The only problem is the type lookup isn't taking the current context path into account.

**Comment by TimDeve** [comment]
Ok, so what you are saying is that declaring my types at the top level should actually be equivalent to the previous behaviour?

**Comment by scolsen** [comment]
Yes, if you declare them outside the module it should work as expected.
